the-collab-lab/tcl-36-smart-shopping-list
# Ld dt firebase setup

## Type of Changes
|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |
## Acceptance Criteria
 [X] Firebase and firestore added to dependencies
 [X] App listens to database changes and displays changes
 [X] Make a change in the app and the data shows up in Firestore
## Related Issue
[Closes #1](https://github.com/the-collab-lab/tcl-36-smart-shopping-list/issues/1)
### Description
This PR adds two new states and functionalities: one for adding items to firestore and another for displaying the items.
### Display items from database
In the App.js we've added a button to handle submitting new items, an input for the user to type data, a button to submit the data from the input field, and a div to display new data from the firebase collection.
You can read more about the firebase setup [here](https://firebase.google.com/docs/firestore/quickstart)
# Testing Steps
- From your terminal, pull down this branch with `git pull origin ld-dt-firebase-setup` and check that branch out with `git checkout ld-dt-firebase-setup`
- Then `yarn` or `npm` install the newly added dependencies locally and `yarn start` or `npm start` to launch the app.
- Fill out the input box with a new item. Make sure it adds it to the div below the input and button and make sure that item appears in the items collection of the database [here](https://console.firebase.google.com/u/0/project/tcl-36-smart-shopping-list/firestore/data/~2Fitems~2Fy5SteQP9oNovRl6WJhhQ).
 
Assignees
@djtaylor8, @lindseyindev
